examples/bri-3/src/bri/ability/ability.factory.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  Ability,
+  ForcedSubject,
+  AbilityClass,
+  AbilityBuilder,
+} from '@casl/ability';
+import { Injectable } from '@nestjs/common';
+import {
+  BpiSubject,
+  BpiSubjectRoleName,
+} from '../identity/bpiSubjects/models/bpiSubject';
+
+const actions = ['manage', 'read'] as const;
+const subjects = ['BpiSubject', 'all'] as const;
+type AppAbilities = [
+  typeof actions[number],
+  (
+    | typeof subjects[number]
+    | ForcedSubject<Exclude<typeof subjects[number], 'all'>>
+  ),
+];
+export type AppAbility = Ability<AppAbilities>;
+export const AppAbility = Ability as AbilityClass<AppAbility>;
+
+type DefinePermissions = (
+  user: BpiSubject,
+  builder: AbilityBuilder<AppAbility>,
+) => void;
+
+const rolePermissions: Record<BpiSubjectRoleName, DefinePermissions> = {
+  [BpiSubjectRoleName.EXTERNAL_BPI_SUBJECT](bpiSubject, { can }) {
+    can('read', 'BpiSubject');
```

Comment:
Should be able to update self right? Or you plan to do this in a separate issue?

Reply:
yea lets do it in separate issue just to keep this small